src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -79,6 +79,20 @@ internal InfiniteScrollList(Lazy<JoinableTaskFactory> joinableTaskFactory)
 
             DataContext = Items;
             CheckBoxesEnabled = false;
+
+            _loadingStatusIndicator.PropertyChanged += LoadingStatusIndicator_PropertyChanged;
+        }
+
+        private void LoadingStatusIndicator_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(LoadingStatusIndicator.Status) )
+            {
+                _joinableTaskFactory.Value.Run(async delegate
```

Comment:
Does this need to happen synchronously?

Reply:
It is not a hard requirement to run synchronously. The requirement is to happen when the loading status is changed. I just tried to follow [VS threading rule 2](https://github.com/microsoft/vs-threading/blob/master/doc/threading_rules.md#rule-2-when-an-implementation-of-an-already-shipped-public-api-must-call-asynchronous-code-and-block-for-its-completion-it-must-do-so-by-following-this-simple-pattern) to switch to the main (UI) thread and update the text, because search actions could happend in background threads.

Reply:
Rule #2 mentions `block for its completion`. 

If we don't need to block for its completion we should do run async.

Reply:
I did that way to avoid dealing with mutual exclusion issues. What's the recommended way to implement mutual exclusion between two async operations ?

Reply:
Is that something you really need to worry about?
How often would we receive this? 
Technically you can't grantee the ordering, but in practice, the tasks are gonna try to switch to the UI thread with the same priority, so the first one should win.

Reply:
> Is that something you really need to worry about?
A little bit. I don't want to inform the user for a "Loading complete" when the packgages list shows a "Loading..." message.

> How often would we receive this?
Each tab change or any change in the search triggers a loading status change.

Let me do a test without JTF to see if it works.

Reply:
It seems to work if I remove JTF syntax, since most of status changes already are in the UI thread. 

I need to verify two more cases and fix a broken unit test:

Case No. 1:

https://github.com/NuGet/NuGet.Client/blob/32cbb0ad13cf4b1abdb961b7ac111463e5f7b31a/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs#L146

Case No: 2:

https://github.com/NuGet/NuGet.Client/blob/32cbb0ad13cf4b1abdb961b7ac111463e5f7b31a/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs#L263

Failing case: DataContext_Initialized_DefaultIsItems

Reply:
I will investigate that later. https://github.com/NuGet/Home/issues/9445